src/App.tsx

```diff
@@ -1,61 +1,159 @@
+    const regexChars = /[\wа-я\s]/ig;
+
+    return titleValue?.match(regexChars)?.join('').trim();
+  };
+
+  const handleFormData = () => {
+    const currentUser = getUser(userName);
+    const id = currentUser?.id;
+    const todosIdCollection = todosArray.map(todo => todo.id);
+    const todosIdMaxValue = Math.max(...todosIdCollection);
+    const modifiedTitle = titleValueModified(titleName);
+    const newTodo: Todo | undefined = {
+      id: todosIdMaxValue + 1,
+      userId: id,
+      title: modifiedTitle,
+      completed: false,
+      user: currentUser,
+    };
+
+    setIsClicked(true);
+
+    if (!modifiedTitle) {
+      setTitleBoolean(true);
+    }
+
+    if (!userName) {
+      setUserBoolean(true);
+    }
+
+    if (userName && modifiedTitle) {
+      setTodosArray(current => [...current, newTodo]);
+      setTitleName('');
+      setUserName('');
+    }
+  };
+
+  const handleUser = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    const { value } = event.target;
+
+    setUserName(value);
+
+    if (isClicked && value) {
+      setUserBoolean(false);
+    }
+  };
+
+  const handleTitle = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const { value } = event.target;
+
+    setTitleName(value);
+
+    if (isClicked && value) {
+      setTitleBoolean(false);
+    }
+  };
 
-export const App = () => {
   return (
     <div className="App">
       <h1>Add todo form</h1>
 
-      <form action="/api/users" method="POST">
+      <form
+        action="/api/users"
+        method="POST"
+        onSubmit={(event) => {
+          event.preventDefault();
+        }}
+      >
         <div className="field">
-          <input type="text" data-cy="titleInput" />
-          <span className="error">Please enter a title</span>
+          <label htmlFor="setTitleName">
+            Title:&nbsp;
+            <input
+              id="setTitleName"
+              type="text"
+              data-cy="titleInput"
+              placeholder="Enter a title"
+              value={titleName}
+              onChange={handleTitle}
+            />
+          </label>
+          {titleBoolean && <span className="error">Please enter a title</span>}
         </div>
 
         <div className="field">
-          <select data-cy="userSelect">
-            <option value="0" disabled>Choose a user</option>
-          </select>
+          <label htmlFor="setUserName">
+            User:&nbsp;
+            <select
+              id="setUserName"
+              data-cy="userSelect"
+              value={userName}
+              onChange={handleUser}
+            >
+              <option value="" disabled>Choose a user</option>
+              {usersFromServer.map(user => (
+                <option
+                  key={user.id}
+                  value={user.name}
+                >
+                  {user.name}
+                </option>
+              ))}
+            </select>
+          </label>
```

Comment:
As I see you use the same logic in a SelectList component
Why didn't you use it?
<img width="1248" alt="image" src="https://user-images.githubusercontent.com/99403198/210068845-32697237-8a56-442c-9182-bf899f6b71f1.png">